app/jobs/bulk_restricted_device_password_emailing_job.rb

```diff
@@ -0,0 +1,47 @@
+class BulkRestrictedDevicePasswordEmailingJob < ApplicationJob
+  queue_as :default
+
+  ALL_SETTINGS = :all
+
+  def perform(rb_limit: ALL_SETTINGS, school_limit: ALL_SETTINGS)
+    @rb_limit = rb_limit
+    @school_limit = school_limit
+    schedule_settings(rbs)
+    schedule_settings(schools)
+  end
+
+private
+
+  attr_reader :rb_limit, :rbs, :school_limit, :schools
+
+  def rb_limit?
+    rb_limit != ALL_SETTINGS
+  end
+
+  def school_limit?
+    school_limit != ALL_SETTINGS
+  end
+
+  def rbs
+    return @rbs if instance_variable_defined?(:@rbs)
+
+    query = ResponsibleBody.with_restricted_devices_and_users
+    @rbs = rb_limit? ? query.limit(rb_limit) : query
```

Comment:
There is a lot going on here too. Maybe we could extract `ResponsibleBody.with_restricted_devices_and_users` to a method and the use the common memoize `@rbs ||=` to make it a bit easier to understand?